fvm/handler/contract.go

```diff
@@ -55,8 +59,15 @@ func (h *ContractHandler) GetContract(address runtime.Address, name string) (cod
 func (h *ContractHandler) SetContract(address runtime.Address, name string, code []byte, signingAccounts []runtime.Address) (err error) {
 	// check if authorized
 	if !h.isAuthorized(signingAccounts) {
-		err = errors.NewOperationAuthorizationErrorf("SetContract", "setting contracts requires authorization from specific accounts")
-		return fmt.Errorf("setting contract failed: %w", err)
+		// check if there's an audit voucher for the contract
+		voucherUsed, err := h.useContractAuditVoucher(address, code)
+		if err != nil {
+			return fmt.Errorf("setting contract failed: (useContractAuditVoucher) %w", err)
+		}
+		if !voucherUsed {
```

Comment:
should it this be if voucher is used we return an error ?

Reply:
maybe renaming the return value to voucherAvailable

Reply:
When a voucher is used, it means that an auditor has approved the contract so we only reject the deployment if no vouchers are used (i.e. no appropriate vouchers exist)

Reply:
Makes sense, addressed here https://github.com/onflow/flow-go/pull/1826/commits/7a550c6655690ba3864d97d61723e258412c3965